### DISCUSSION TITLE
Aditional arguments to objective function

### DISCUSSION THREAD

**Discussion by rogerfn** [PasaOpasen/geneticalgorithm2-6.8.7#19] [Show and tell]
Thank you for this very nice package.

I think it would be great to pass arguments to the objective function. 

There are some cases where I need to pass an object to the objective function and unfortunately the only way to do that at the moment is to define as a global variable.

I could make a PR for that if necessary

**Comment by PasaOpasen** [comment]
@rogerfn make PR if it's possible to u

So I did't understand what exactly u want

Do u want to use logic like

```python

g = 0


def f(x):
    global g
    g += 1
    return x+g

model = ga(f) 
```

?

Show me an example how do u plan to use the package with pass arguments

**Reply by rogerfn** [reply]
I was thinking in something like the the interface that scipy.optimize.minimize has:

```python
def f(x, a, b):
  return a*x + b
```
and in the ga function you would have something like:
```python
ga(f, arguments=(a,b))
```

**Reply by PasaOpasen** [reply]
@rogerfn and u want to change `a, b` during optimization process???

Why u don't use

```python
def f(arr):
   return arr[1]*arr[0] + arr[2]

```

**Comment by rogerfn** [comment]
No, those are constant objects.

**Reply by PasaOpasen** [reply]
@rogerfn So why u cannot create carrier function like

```python
f_local = lambda x: f(x,a,b)

model = ga(f_local)

```

**Comment by rogerfn** [comment]
Yes, actually that’s what I’m doing. It is mainly have the same or similar
interface as scipy
